homeassistant/components/shelly/__init__.py

```diff
@@ -290,6 +290,11 @@ async def _async_setup_rpc_entry(hass: HomeAssistant, entry: ShellyConfigEntry)
         )
         runtime_data.rpc = ShellyRpcCoordinator(hass, entry, device)
         runtime_data.rpc.async_setup(runtime_data.platforms)
+        # Try to connect to the device, if we reached here from config flow
+        # and user woke up the device when adding it, we can continue setup
+        # otherwise we will wait for the device to wake up
+        if sleep_period:
```

Comment:
We can remove this condition on a cleanup PR which will remove handling for `None` as sleep period https://github.com/home-assistant/core/blob/9e762fa2223bd4c1610785108b7afa29f220b292/homeassistant/components/shelly/__init__.py#L280 since we already for it to `int`  - https://github.com/home-assistant/core/blob/9e762fa2223bd4c1610785108b7afa29f220b292/homeassistant/components/shelly/utils.py#L292-L294

This is a leftover from legacy code that did not store the sleep period but it is fixed for few years already.